cmd/snap-seccomp/main_test.go

```diff
@@ -0,0 +1,404 @@
+		{"setns - CLONE_NEWUTS", "setns;native;0,99", main.SeccompRetKill},
+		// test_restrictions_working_args_mkdnor
+		{"mknod - S_IFREG", "mknod;native;0,S_IFREG", main.SeccompRetAllow},
+		{"mknod - S_IFCHR", "mknod;native;0,S_IFCHR", main.SeccompRetAllow},
+		{"mknod - S_IFBLK", "mknod;native;0,S_IFBLK", main.SeccompRetAllow},
+		{"mknod - S_IFIFO", "mknod;native;0,S_IFIFO", main.SeccompRetAllow},
+		{"mknod - S_IFSOCK", "mknod;native;0,S_IFSOCK", main.SeccompRetAllow},
+		{"mknod - S_IFREG", "mknod;native;0,999", main.SeccompRetKill},
+		{"mknod - S_IFCHR", "mknod;native;0,999", main.SeccompRetKill},
+		{"mknod - S_IFBLK", "mknod;native;0,999", main.SeccompRetKill},
+		{"mknod - S_IFIFO", "mknod;native;0,999", main.SeccompRetKill},
+		{"mknod - S_IFSOCK", "mknod;native;0,999", main.SeccompRetKill},
+		// test_restrictions_working_args_prio
+		{"setpriority PRIO_PROCESS", "setpriority;native;PRIO_PROCESS", main.SeccompRetAllow},
+		{"setpriority PRIO_PGRP", "setpriority;native;PRIO_PGRP", main.SeccompRetAllow},
+		{"setpriority PRIO_USER", "setpriority;native;PRIO_USER", main.SeccompRetAllow},
+		{"setpriority PRIO_PROCESS", "setpriority;native;99", main.SeccompRetKill},
+		{"setpriority PRIO_PGRP", "setpriority;native;99", main.SeccompRetKill},
+		{"setpriority PRIO_USER", "setpriority;native;99", main.SeccompRetKill},
+		// test_restrictions_working_args_termios
+		{"ioctl - TIOCSTI", "ioctl;native;0,TIOCSTI", main.SeccompRetAllow},
+		{"ioctl - TIOCSTI", "quotactl;native;0,99", main.SeccompRetKill},
+	} {
+		simulateBpf(c, t.seccompWhitelist, t.bpfInput, t.expected)
+	}
+}
+
+func (s *snapSeccompSuite) TestCompileBadInput(c *C) {
+	for _, t := range []struct {
+		inp    string
+		errMsg string
+	}{
+		// test_bad_seccomp_filter_args_clone (various typos in input)
+		{"setns - CLONE_NEWNE", `cannot parse line: cannot parse token "CLONE_NEWNE" \(line "setns - CLONE_NEWNE"\)`},
+		{"setns - CLONE_NEWNETT", `cannot parse line: cannot parse token "CLONE_NEWNETT" \(line "setns - CLONE_NEWNETT"\)`},
+		{"setns - CL0NE_NEWNET", `cannot parse line: cannot parse token "CL0NE_NEWNET" \(line "setns - CL0NE_NEWNET"\)`},
+		// test_bad_seccomp_filter_args_mknod (various typos in input)
+		{"mknod - |S_IFIF", `cannot parse line: cannot parse token "S_IFIF" \(line "mknod - |S_IFIF"\)`},
+		{"mknod - |S_IFIFOO", `cannot parse line: cannot parse token "S_IFIFOO" \(line "mknod - |S_IFIFOO"\)`},
+		{"mknod - |S_!FIFO", `cannot parse line: cannot parse token "S_IFIFO" \(line "mknod - |S_!FIFO"\)`},
+		// test_bad_seccomp_filter_args_null
+		{"socket S\x00CK_STREAM", `cannot parse line: cannot parse token .*`},
+		{"socket SOCK_STREAM\x00bad stuff", `cannot parse line: cannot parse token .*`},
+
+		// test_bad_seccomp_filter_args
+		{"mbind - - - - - - 7", `cannot parse line: too many arguments specified for syscall 'mbind' in line.*`},
+		{"mbind 1 2 3 4 5 6 7", `cannot parse line: too many arguments specified for syscall 'mbind' in line.*`},
```

Comment:
You are missing the `for i in 'bar' '-1' '0 - -1 0' '--10' '0:10' '1-10' '0,1' '0x0' 'a1' '1a' '1-' '1\\n1' '1\ 2' '999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999' ; do` tests from test_bad_seccomp_filter_args.